code_samples/back_office/search/src/Search/SortingDefinition/Provider/RandomSortingDefinitionProvider.php

```diff
@@ -0,0 +1,41 @@
+<?php declare(strict_types=1);
+
+namespace App\Search\SortingDefinition\Provider;
+
+use Ibexa\Contracts\Core\Repository\Values\Content\Query\SortClause;
+use Ibexa\Contracts\Search\SortingDefinition\SortingDefinition;
+use Ibexa\Contracts\Search\SortingDefinition\SortingDefinitionProviderInterface;
+use JMS\TranslationBundle\Model\Message;
+use JMS\TranslationBundle\Translation\TranslationContainerInterface;
+use Symfony\Contracts\Translation\TranslatorInterface;
+
+final class RandomSortingDefinitionProvider implements SortingDefinitionProviderInterface, TranslationContainerInterface
+{
+    private TranslatorInterface $translator;
+
+    public function __construct(TranslatorInterface $translator)
+    {
+        $this->translator = $translator;
+    }
+
+    public function getSortingDefinitions(): array
+    {
+        return [
+            new SortingDefinition(
+                'random',
+                $this->translator->trans('sort_definition.random.label'),
+                [
+                    new SortClause\Random(),
```

Comment:
TODO: Find another example, `SortClause\Random` can't be used, see https://github.com/ibexa/search/pull/35#pullrequestreview-1740417307